src/Core/Managed/Shared/DataContracts/TelemetryContext.cs

```diff
@@ -40,6 +41,7 @@ internal TelemetryContext(IDictionary<string, string> properties)
             Debug.Assert(properties != null, "properties");
             this.properties = properties;
             this.tags = new ConcurrentDictionary<string, string>();
+            this.correlationContext = new ConcurrentDictionary<string, string>();
```

Comment:
Actually, I really like the idea of ```ImmutableDictionary<K,V>``` because they're zero-cost to copy which will help in ```StartOperation```
However, I know @SergeyKanzhelev has previously expressed a desire not to include additional dependencies.